qiskit/opflow/converters/circuit_sampler.py

```diff
@@ -172,20 +177,93 @@ def convert(
             if self._caching == "last":
                 self.clear_cache()
 
-            # convert to circuit and reduce
-            operator_dicts_replaced = operator.to_circuit_op()
-            self._reduced_op_cache = operator_dicts_replaced.reduce()
-
-            # extract circuits
-            self._circuit_ops_cache = {}
-            self._extract_circuitstatefns(self._reduced_op_cache)
-            if not self._circuit_ops_cache:
-                raise OpflowError(
-                    "Circuits are empty. "
-                    "Check that the operator is an instance of CircuitStateFn or its ListOp."
-                )
-            self._transpiled_circ_cache = None
-            self._transpile_before_bind = True
+            if (
+                self._split_transpile
+                and isinstance(operator, ComposedOp)
+                and len(operator) == 2
+                and isinstance(operator[1], CircuitStateFn)
+            ):
+                self._reduced_op_cache = operator.to_circuit_op().reduce()
+                self._circuit_ops_cache = {}
+                self._extract_circuitstatefns(self._reduced_op_cache)
+                if not self._circuit_ops_cache:
+                    raise OpflowError(
+                        "Circuits are empty. "
+                        "Check that the operator is an instance of CircuitStateFn or its ListOp."
+                    )
+                circuits = [
+                    op_c.to_circuit(meas=not self._statevector)
+                    for op_c in list(self._circuit_ops_cache.values())
+                ]
+
+                common_circuit = operator[1].to_circuit_op().reduce().to_circuit()
+                try:
+                    # 1. transpile a common circuit
+                    transpiled_common_circuit = self.quantum_instance.transpile(common_circuit)[0]
+
+                    # 2. transpile diff circuits
+                    diff_circuits = []
+                    for circuit in circuits:
+                        diff_circuit = circuit.copy()
+                        del diff_circuit.data[0 : len(common_circuit)]
+                        diff_circuits.append(diff_circuit)
+
+                    if transpiled_common_circuit._layout is not None:
+                        layout = transpiled_common_circuit._layout.copy()
```

Comment:
I suppose we need to set the "final" layout here, however, `_layout` stores the initial layout of the transpiled common circuit.

Unfortunately, we have no simple way to get the final layout of a transpiled circuit so far.
The following could work (I know it's not smart. Does anyone think of any better idea?):
- Adding dummy measures to the common circuit before transpiling
- Constructing the final layout from virtuals-clbits and physical-clbits maps stored in original and transpiled measures respectively
- Removing the dummy measures from the common circuit

Reply:
Thank you. You are right. We assume `_layout` is the final layout.
I have fixed the PR from your advice. Is this correct? 7418ac2

Reply:
I have shown how one can get the required final layout (permutation) here: https://github.com/Qiskit/qiskit-terra/pull/6827

Reply:
Thank you so much. It's really useful.